### PR TITLE
prompting: introduce baseline prompting infrastructure

### PR DESCRIPTION
**Pull request by olivercalder**
This lays the groundwork for apparmor prompting. Most notably, this includes the notifier and other apparmor-related infrastructure for handling prompt messages from the kernel, and the a means of enabling prompting with vendored apparmor for particular interfaces. Prompting is added to the `home` interface in this changelist.

### CODE REVIEW

File: osutil/epoll/epoll.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+			e.Close()
+		}
+	})
+	return e, nil
+}
+
+// Close closes the event monitoring descriptor.
+func (e *Epoll) Close() error {
+	runtime.SetFinalizer(e, nil)
+	return syscall.Close(e.fd)
```
Comment (mvo5) [inline osutil/epoll/epoll.go:89]: This should probably also set `e.fd = -1`, no ?

Comment (olivercalder) [reply osutil/epoll/epoll.go:89]: Good catch, thanks, I'll confirm and make the change (in a different PR).

**Comment by olivercalder** [comment]
I'm going to close this PR and then reopen smaller incremental ones to make these changes more manageable.

**Comment by olivercalder** [comment]
I'm going to close this PR and then reopen smaller incremental ones to make these changes more manageable.

**Pull request closed without merging**